src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -0,0 +1,373 @@
+                    && m.methodName().equalsString(INIT_NAME)) {
+                errors.add(new VerifyError("Interface cannot have a method named <init> in %s".formatted(toString(classModel))));
+            }
+            verifyMethodName(m.methodName().stringValue());
+        } catch (VerifyError ve) {
+            errors.add(ve);
+        }
+    }
+
+    private void verifyAttributes(ClassFileElement cfe, List<VerifyError> errors) {
+        if (cfe instanceof AttributedElement ae) {
+            var attrNames = new HashSet<String>();
+            for (var a : ae.attributes()) {
+                if (!a.attributeMapper().allowMultiple() && !attrNames.add(a.attributeName())) {
+                    errors.add(new VerifyError("Multiple %s attributes in %s".formatted(a.attributeName(), toString(ae))));
+                }
+                verifyAttribute(ae, a, errors);
+            }
+        }
+        switch (cfe) {
+            case CompoundElement<?> comp -> {
+                for (var e : comp) verifyAttributes(e, errors);
+            }
+            case RecordAttribute ra -> {
+                for(var rc : ra.components()) verifyAttributes(rc, errors);
+            }
+            default -> {}
+        }
+    }
+
+    private void verifyAttribute(AttributedElement ae, Attribute<?> a, List<VerifyError> errors) {
+        int payLoad = ((BoundAttribute)a).payloadLen();
+        if (payLoad != switch (a) {
+            case AnnotationDefaultAttribute aa -> valueSize(aa.defaultValue());
+            case BootstrapMethodsAttribute bma -> {
+                int l = 2;
+                for (var bm : bma.bootstrapMethods()) {
+                    l += 4 + 2 * bm.arguments().size();
+                }
+                yield l;
+            }
+            case CharacterRangeTableAttribute cra -> 2 + 14 * cra.characterRangeTable().size();
+            case CodeAttribute ca -> {
+                MethodModel mm = (MethodModel)ae;
+                if (mm.flags().has(AccessFlag.NATIVE) || mm.flags().has(AccessFlag.ABSTRACT)) {
+                    errors.add(new VerifyError("Code attribute in native or abstract %s".formatted(toString(ae))));
+                }
+                if (ca.maxLocals() < Util.maxLocals(mm.flags().flagsMask(), mm.methodTypeSymbol())) {
+                    errors.add(new VerifyError("Arguments can't fit into locals in %s".formatted(toString(ae))));
+                }
+                int l = 12 + ca.codeLength() + 8 * ca.exceptionHandlers().size();
+                for (var caa : ca.attributes()) {
+                    l += 6 + ((BoundAttribute)caa).payloadLen();
+                }
+                yield l;
+            }
+            case CompilationIDAttribute cida -> 2;
+            case ConstantValueAttribute cva -> {
+                ClassDesc type = ((FieldModel)ae).fieldTypeSymbol();
+                ConstantValueEntry cve = cva.constant();
+                if (!switch (TypeKind.from(type)) {
```

Comment:
Weird-looking switch

Reply:
It may be (and actually was) composed into a single boolean expression, however effectivity and readability was worse. Boolean switch expression across Enum seems to me effective and clear to read.